AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/UserOperations/AWSAuthUpdateUserAttributeOperation.swift

```diff
@@ -0,0 +1,144 @@
+    
+    private let authStateMachine: AuthStateMachine
+    private let credentialStoreStateMachine: CredentialStoreStateMachine
+    private let userPoolFactory: CognitoUserPoolFactory
+    private var statelistenerToken: AuthStateMachineToken?
+    private let fetchAuthSessionHelper: FetchAuthSessionOperationHelper
+    
+    init(_ request: AuthUpdateUserAttributeRequest,
+         authStateMachine: AuthStateMachine,
+         credentialStoreStateMachine: CredentialStoreStateMachine,
+         userPoolFactory: @escaping CognitoUserPoolFactory,
+         resultListener: ResultListener?)
+    {
+        self.authStateMachine = authStateMachine
+        self.credentialStoreStateMachine = credentialStoreStateMachine
+        self.userPoolFactory = userPoolFactory
+        self.fetchAuthSessionHelper = FetchAuthSessionOperationHelper(
+            authStateMachine: authStateMachine,
+            credentialStoreStateMachine: credentialStoreStateMachine)
+        super.init(categoryType: .auth,
+                   eventName: HubPayload.EventName.Auth.fetchUserAttributesAPI,
+                   request: request,
+                   resultListener: resultListener)
+    }
+    
+    override public func main() {
+        if isCancelled {
+            finish()
+            return
+        }
+        
+        fetchAuthSessionHelper.fetchSession { [weak self] result in
+            switch result {
+            case .success(let session):
+                guard let cognitoTokenProvider = session as? AuthCognitoTokensProvider,
+                      let tokens = try? cognitoTokenProvider.getCognitoTokens().get() else {
+                    self?.dispatch(AuthError.unknown("Unable to fetch auth session", nil))
+                    return
+                }
+                Task.init { [weak self] in
+                    await self?.updateAttribute(with: tokens.accessToken)
+                }
+            case .failure(let error):
+                self?.dispatch(error)
+            }
+        }
+        
+    }
+    
+    func updateAttribute(with accessToken: String) async {
+        
+        do {
+            let userPoolService = try userPoolFactory()
+            
+            let attributeList = [request.userAttribute]
+            
+            let clientMetaData = (request.options.pluginOptions as? AWSUpdateUserAttributeOptions)?.metadata ?? [:]
+            let input = UpdateUserAttributesInput(accessToken: accessToken,
+                                                  clientMetadata: clientMetaData,
+                                                  userAttributes: attributeList.map({ $0.sdkClientAttributeType() }))
+            
+            let result = try await userPoolService.updateUserAttributes(input: input)
+            
+            if self.isCancelled {
+                return
+            }
+            
+            guard let codeDeliveryDetailsList = result.codeDeliveryDetailsList else {
```

Comment:
Does the service always return this value? What about cases where we do not need to verify the update?

Reply:
So  the list is an optional that's why I need to place a guard around it. 
```
public var codeDeliveryDetailsList: [CognitoIdentityProviderClientTypes.CodeDeliveryDetailsType]?
```

In cases where the next step is not needed, the list would be empty and not `nil`. In those scenarios, the for loop underneath should handle it. Which is as follows. It iterates the dictionary and if the attribute doesn't exist, will update the result accordingly. 

```
            // Check if all items are added to the dictionary
            for item in attributeList where finalResult[item.key] == nil {
                let updateAttributeResult = AuthUpdateAttributeResult(isUpdated: true, nextStep: .done)
                finalResult[item.key] = updateAttributeResult
            }
```